pkg/scheduler/core/generic_scheduler.go

```diff
@@ -936,15 +936,18 @@ func filterPodsWithPDBViolation(pods []*v1.Pod, pdbs []*policy.PodDisruptionBudg
 				if selector.Empty() || !selector.Matches(labels.Set(pod.Labels)) {
 					continue
 				}
+
+				// Existing in DisruptedPods means it has been processed in API server,
+				// we don't treat it as a violating case.
+				if _, exist := pdb.Status.DisruptedPods[pod.Name]; exist {
+					continue
+				}
+				// Only decrement the matched pdb when it's not in its <DisruptedPods>;
+				// otherwise we may over-decrement the budget number.
+				pdbsAllowed[i]--
```

Comment:
wouldn't this make the value go into the minus? I guess it doesn't matter if that happens, right?

Reply:
Yes, pdbsAllowed[i] can be negative now.

Another style is like this:

```go
if pdbsAllowed[i] <= 0 {
    pdbForPodIsViolated = true
} else {
    pdbsAllowed[i]--
}
```

Not quite make a big difference I think :)